### PR TITLE
Fix a bug in createTextNode when the data argument is a non-string

### LINKED ISSUE #1380: `data` argument of `createTextNode` is not coerced into a string
**Describe the bug**

Hello! I'm filing this issue in response to an issue that got raised over at https://github.com/testing-library/svelte-testing-library/issues/343.

- `document.createTextNode(data)` is used to create text nodes
- The input `data` is expected to be a DOMString
- In browsers I've tested (latest Chromium, Firefox, and Safari), if you pass a **non-string** to `createTextNode`, it will be coerced into a string
- In Happy DOM, the data is not coerced, and `node.data` ends up _not_ a string, causing problems in downstream consumers

**To Reproduce**

```ts
// index.js
import { Window } from "happy-dom";

const window = new Window({ url: "https://localhost:8080" });
const node = window.document.createTextNode(2);

console.log(node.data, typeof node.data);
```

```shell
> node index.js
2 number
```

In the case of the issue filed in `@testing-library/svelte`: `svelte@4` erroneously passes a non-string to `createTextNode`, and then the pretty printing library that is used to print test results chokes because it tries to do a `String.replace` on `node.data`, which may not be a string

**Expected behavior**

I expect the resulting `data` property of `createTextNode` to always be a string

**Additional context**

Tested with `happy-dom@14.4.0`

### PR DESCRIPTION
**Pull request by odanado**
Close #1380

Run in Chrome:

Run in Safari:

Run in Firefox:

### CODE REVIEW

File: packages/happy-dom/src/nodes/document/Document.ts [hunk 1]
```diff
@@ -1165,8 +1165,13 @@ export default class Document extends Node {
 	 * @param [data] Text data.
 	 * @returns Text node.
 	 */
-	public createTextNode(data?: string): Text {
-		return NodeFactory.createNode<Text>(this, this[PropertySymbol.ownerWindow].Text, data);
+	public createTextNode(data: string): Text {
```
Comment (odanado) [inline packages/happy-dom/src/nodes/document/Document.ts:1168]: In TypeScript `lib.dom.d.ts`, the createTextNode argument is not optional.




I have changed the type and implementation to mimic browser behavior.

| Safari | Firefox | Chrome |
|-|-|-|
||||

Could you help me determine whether this change is considered a 'bug fix' or a 'breaking change'?

**Comment by odanado** [comment]
```
   FAIL  test/fetch/SyncFetch.test.ts > SyncFetch > send() > Updates cache after a failed revalidation with a "If-Modified-Since" request for a GET response with "Cache-Control" set to a "max-age".
  AssertionError: expected 'some text' to be 'some new text' // Object.is equality
```

I noticed that the CI tests failed after I submitted my pull request. However, the failure appears to be related to a test that is not associated with the changes I've made. I suspect it might be a flaky test. Could you please advise on how to proceed? Should I try rerunning the tests, or is there anything else you'd recommend?

**Review by capricorn86 — APPROVED** [review]
Thank you for your contribution @odanado! :star:

**Comment by capricorn86** [comment]
> ```
>    FAIL  test/fetch/SyncFetch.test.ts > SyncFetch > send() > Updates cache after a failed revalidation with a "If-Modified-Since" request for a GET response with "Cache-Control" set to a "max-age".
>   AssertionError: expected 'some text' to be 'some new text' // Object.is equality
> ```
> 
> I noticed that the CI tests failed after I submitted my pull request. However, the failure appears to be related to a test that is not associated with the changes I've made. I suspect it might be a flaky test. Could you please advise on how to proceed? Should I try rerunning the tests, or is there anything else you'd recommend?

It's a known problem that has to be fixed. It is related to a timing issue with the cache expire time for in `Fetch.ts` and `SyncFetch.ts`.

**Pull request merged**